detectable/src/main/java/com/blackduck/integration/detectable/detectables/yarn/YarnTransformer.java

```diff
@@ -210,42 +212,68 @@ private LazyBuilderMissingExternalIdHandler getLazyBuilderHandler(List<NameVersi
             if (externalId.isPresent()) {
                 return externalId.get();
             } else {
-                ExternalId lazilyGeneratedExternalId;
-                if (!unMatchedDependencies.contains(dependencyId)) {
-                    logger.warn("Unable to find standard NPM package identification details for '{}' in the yarn.lock file", dependencyId);
-                    unMatchedDependencies.add(dependencyId);
+                Optional<NameVersion> nameVersionOptional;
+                if (!unMatchedDependencies.containsKey(dependencyId)) {
+                    String dependencyIdClean = dependencyId.toString().replace("\\/","/");
+                    nameVersionOptional = getNameVersion(dependencyIdClean);
+                    logger.warn("No standard NPM package identification details are available for '{}' from the yarn.lock file", dependencyIdClean);
```

Comment:
Improved warning message about lack of standard NPM details to resolve remaining unmatched components.